op3_ball_detector/src/ball_detector.cpp

```diff
@@ -342,6 +343,23 @@ bool BallDetector:: getParamCallback(op3_ball_detector::GetParameters::Request &
   return true;
 }
 
+// Callback for image saving
+bool BallDetector::saveImageCallback(op3_ball_detector::SaveImage::Request &req, op3_ball_detector::SaveImage::Response &res)
+{
+  std::string filename = "/home/robotis/" + req.params.name + ".png";
+  if (cv_img_ptr_sub_ != NULL) {
+    if (cv::imwrite(filename, cv_img_ptr_sub_->image) == false) {
+      res.returns.name = "Failed for some reason";
+    }
+    res.returns.name = "Saved";
```

Comment:
Wouldn't this overwrite line 352 or am I misunderstanding something?

Reply:
9626c6d fixes the omitted else statement.